homeassistant/components/synology_dsm/__init__.py

```diff
@@ -590,21 +599,21 @@ def __init__(
         self,
         api: SynoApi,
         entity_type: str,
-        entity_info: dict[str, str],
+        entity_info: dict[str, Any],
```

Comment:
Or define proper `TypedDict`.
```suggestion
        entity_info: dict[str, str | bool | None],
```

Reply:
When i try this, I get additional erros :confused: 
```
homeassistant/components/synology_dsm/__init__.py:616: error: Incompatible types in assignment (expression has type "Union[bool, str, None]", variable has type "Optional[str]")  [assignment]
homeassistant/components/synology_dsm/__init__.py:617: error: Incompatible types in assignment (expression has type "Union[bool, str, None]", variable has type "bool")  [assignment]
homeassistant/components/synology_dsm/__init__.py:618: error: Incompatible types in assignment (expression has type "Union[bool, str, None]", variable has type "Optional[str]")  [assignment]
homeassistant/components/synology_dsm/__init__.py:619: error: Incompatible types in assignment (expression has type "Union[bool, str, None]", variable has type "Optional[str]")  [assignment]
```

Reply:
Create `TypedDict` for `entity_info`:

```python
class EntityInfo(TypedDict):
  name: str
  unit: str | None
  icon: str | None
  class: str | None
  enable: bool
```